### ISSUE TITLE
|trans twig filter does not work

### ISSUE THREAD

**Issue by webBook-cz** [OFFLINE-GmbH/oc-mall-plugin#1011]
Hi! I use translation strings in my theme (https://docs.octobercms.com/3.x/cms/themes/localization.html#localization-file-structure)
When your plugin is installed the twig filter |trans does not work anymore.

**Comment by tobias-kuendig** [comment]
What does "not work anymore" mean in that case?

Do you see any errors? Anything in the logs?

**Comment by webBook-cz** [comment]
No,  I do not see any errors, just the text is not translated.

**Comment by tobias-kuendig** [comment]
Can you share the version details of you setup? (October, Mall)

**Comment by webBook-cz** [comment]
Hi Tobias!
try it here please: http://eshop.forfree.cz/backend
login: Tobias
pw: t0b145

on page http://eshop.forfree.cz/en/test are 2 translated vars waiting and client (http://eshop.forfree.cz/cs/test )
If you swich-off your plugin the page works like a charm...

**Comment by webBook-cz** [comment]
Any chance to solve it?

**Comment by tobias-kuendig** [comment]
I'd have to dig into this myself, but I currently do not have the time. 

The problem probably comes from around here: https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/classes/registration/BootTwig.php#L32

It probably is a RainLab.Translate 2.0 incompatibility. October does register the `_` and `__` filters in Version 3.0 too (in modules/system/twig/Extension.php#L93), so there might be a conflict.

**Comment by tobias-kuendig** [comment]
I see that the problem occurs on your site, but I cannot reproduce it locally. Could you share your translation strings? Are they located in `themes/xyz/lang/xx.json`?

**Comment by webBook-cz** [comment]
Yes, exactly

**Comment by chocolata** [comment]
Hi guys, just upgraded my testenvironment (October CMS v3.3.17, Mall v3.0.18, Translate v2.2.6)  and am also experiencing issues with the translations. In my case the translations reside in the lang folder at root level:

![image](https://github.com/OFFLINE-GmbH/oc-mall-plugin/assets/3584586/83dbe75e-ba47-4add-8256-07f9326feb1b)

The content of the lang.php file for example is the following:

```
<?php

return [
    'components.productsFilter.sortOrder.propertyValues.15' => 'Datum',
    'general_settings.catalogue' => 'Catalogus',
    'components.productsFilter.sortOrder.date' => 'Datum',
    'frontend.stock.checking' => 'Voorraad controleren...',
    'frontend.stock.available' => 'Één artikel op voorraad|:stock op voorraad',
    'stock.unavailable' => 'Geen voorraad meer',
    'stock.many_available' => 'Op voorraad',
    'common.yes' => '✔',
    'common.no' => '✘',
    'reviews.title' => 'Titel',
    'reviews.review' => 'Details',
    'reviews.pros' => 'Goede punten (een per regel)',
    'reviews.cons' => 'Slechte punten (een per regel)',
    'components.productsFilter.sortOrder.bestseller' => 'Best verkocht',
    'components.productsFilter.sortOrder.priceLow' => 'Laagste prijs',
    'components.productsFilter.sortOrder.priceHigh' => 'Hoogste prijs',
    'components.productsFilter.sortOrder.latest' => 'Meest recent',
    'components.productsFilter.sortOrder.oldest' => 'Oudste',
    'components.productsFilter.sortOrder.random' => 'Willekeurig',
    'components.productsFilter.sortOrder.manual' => 'Standaard',
    'components.productsFilter.sortOrder.name' => 'Naam',
    'components.productsFilter.sortOrder.ratings' => 'Best beoordeeld',
    'common.catalogue' => 'Aanbod',
    'common.orders' => 'Bestellingen',
    'common.discounts' => 'Kortingen',
    'common.payments' => 'Betalingen',
    'common.products' => 'Producten',
    'common.categories' => 'Categorieën',
    'common.brands' => 'Merken',
    'common.properties' => 'Eigenschappen',
    'common.services' => 'Diensten',
];
```
It seems like the translations are not taken into account at all and just revert to their defaults. 

I understand some configuration is required after migrating to the Rainlab.Translate v2 plugin, and I think I've followed all the steps. I've set he language code of the primary site to the "nl" language code, and set the language dropdown accordingly:
![image](https://github.com/OFFLINE-GmbH/oc-mall-plugin/assets/3584586/d5bc0471-900f-4cb1-9e8a-0145ea7b8e13)

It might not be relevant, but my back-end environment is also set to the "nl" language code.

What is the recommended way of customizing the language strings in combination with the Translate v2 plugin? 

As a workaround, could we temporarily downgrade to v1.12 for example? I see that the current version of the Mall plugin required V2 of the translate plugin.

Thanks for looking into this.

**Comment by tobias-kuendig** [comment]
@chocolata from the OCMS v3 Upgrade guide:

> The lang directory for custom translation overrides has moved to app/lang

You have to move your `lang` folder in a `app` folder. Does the issue still persist?

https://octobercms.com/support/article/rn-30#localization-updates

**Comment by tobias-kuendig** [comment]
@webBook-cz Just to be sure, this couldn't be the same problem on your side? I am still unable to reproduce the issue.

**Comment by webBook-cz** [comment]
no, The traslation is in theme folder:  themes/mytheme/lang/en.json - and without Mall plugin works perfect.

**Comment by tobias-kuendig** [comment]
I'm closing this issue due to inactivity. If you need any further assistance feel free to reply.

**Comment by webBook-cz** [comment]
And is it solved?
